pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,263 @@
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to create user settings: %v", err)})
+		} else {
+			serverutils.SendResponse(w, http.StatusOK, configMap)
+		}
+	case http.MethodDelete:
+		err := h.deleteUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to delete user settings: %v", err)})
+		} else {
+			w.WriteHeader(http.StatusNoContent)
+		}
+	default:
+		w.Header().Set("Allow", "GET, POST, DELETE")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Unsupported method, supported methods are GET POST DELETE"})
+	}
+}
+
+// Fetch the current user with auth.User token and returning the user-setting ConfigMap.
+func (h *UserSettingsHandler) getUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Get(context, resourceData.getConfigMapName(), meta.GetOptions{})
+}
+
+// Fetch the current user with auth.User token and create a new user-setting ConfigMap.
+// Returns the existing ConfigMap if it is already exist.
+func (h *UserSettingsHandler) createUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	role := &rbac.Role{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "Role",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleName(),
+		},
+		Rules: []rbac.PolicyRule{
+			rbac.PolicyRule{
+				APIGroups: []string{
+					"", // Core group, not "v1"
+				},
+				Resources: []string{
+					"configmaps", // Not "ConfigMap"
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"patch",
+					"update",
+					"watch",
+				},
+				ResourceNames: []string{
+					resourceData.getConfigMapName(),
+				},
+			},
+		},
+	}
+
+	roleBinding := &rbac.RoleBinding{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "RoleBinding",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleBindingName(),
+		},
+		Subjects: []rbac.Subject{
+			rbac.Subject{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "User",
+				Name:     user.Username,
+			},
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     resourceData.getRoleName(),
+		},
+	}
+
+	configMap := &core.ConfigMap{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getConfigMapName(),
+		},
+	}
+
+	client, err := h.createServiceAccountClient()
```

Comment:
Also this call can be moved to the `HandleUserSettings()` and pas as param to the appropriate `UserSettingsHandler` method.

Reply:
Done